### PR TITLE
add gzip compression

### PR DESCRIPTION
**Pull request by Oded-R**

### CODE REVIEW

File: src/Core/HttpClient/IHttpClient.cs [hunk 1]
```diff
@@ -9,27 +10,41 @@ namespace Logzio.DotNet.Core.WebClient
...
     public class HttpClientHandler : IHttpClient
     {
-        private readonly HttpClient _client = new HttpClient();
+        private readonly HttpClient _client;
         private static readonly System.Text.Encoding _encodingUtf8 = new System.Text.UTF8Encoding(false);
-        private readonly MediaTypeHeaderValue _headerValue = new MediaTypeHeaderValue("application/json") { CharSet = _encodingUtf8.WebName };
-       
+        private readonly MediaTypeHeaderValue _headerCompressed = new MediaTypeHeaderValue("text/html") { CharSet = _encodingUtf8.WebName };
+        private readonly MediaTypeHeaderValue _headerJson = new MediaTypeHeaderValue("text/html") { CharSet = _encodingUtf8.WebName };
+
```
Comment (idohalevi) [inline src/Core/HttpClient/IHttpClient.cs]: @Oded-R 
Why did you change the header? is it a must for a compressed body?

Comment (Oded-R) [reply src/Core/HttpClient/IHttpClient.cs]: you are right they both work with "application/json"
I will fix it now.
